docs/side_quests/nf-test.md

```diff
@@ -0,0 +1,1074 @@
+
+        when {
+            params {
+                // define parameters here. Example:
+                // outdir = "tests/results"
+            }
+        }
+
+        then {
+            assert workflow.success
+        }
+
+    }
+
+}
+```
+
+We'll take a second to understand the structure of the test file.
+
+The `nextflow_pipeline` block is the entry point for all pipeline level tests. It contains the following:
+
+- `name`: The name of the test.
+- `script`: The path to the pipeline script.
+
+The `test` block is the actual test. It contains the following:
+
+- `when`: The conditions under which the test should be run. This includes the parameters that will be used to run the pipeline.
+- `then`: The assertions that should be made. This includes the expected outcomes of the pipeline.
+
+In plain English, the logic of the test reads as follows:
+"**When** these _parameters_ are provided to this _pipeline_, **then** we expect to see these results."
+
+This isn't a functional test, we will demonstrate how to turn it into one in the next section.
+
+### A Note on Test Names
+
+In the example above, we used the default name "Should run without failures" which is appropriate for a basic test that just checks if the pipeline runs successfully. However, as we add more specific test cases, we should use more descriptive names that indicate what we're actually testing. For example:
+
+- "Should convert input to uppercase" - when testing specific functionality
+- "Should handle empty input gracefully" - when testing edge cases
+- "Should respect max memory parameter" - when testing resource constraints
+- "Should create expected output files" - when testing file generation
+
+Good test names should:
+
+1. Start with "Should" to make it clear what the expected behavior is
+2. Describe the specific functionality or scenario being tested
+3. Be clear enough that if the test fails, you know what functionality is broken
+
+As we add more assertions and specific test cases later, we'll use these more descriptive names to make it clear what each test is verifying.
+
+## 1.2. Run the test
+
+Let's run the test to see what happens.
+
+```bash
+nf-test test tests/main.nf.test
+```
+
+```console title="nf-test pipeline fail"
+> nf-test test tests/main.nf.test
+
+🚀 nf-test 0.9.2
+https://www.nf-test.com
+(c) 2021 - 2024 Lukas Forer and Sebastian Schoenherr
+
+
+Test Workflow main.nf
+
+  Test [693ba951] 'Should run without failures' FAILED (4.652s)
+
+  Assertion failed:
+
+  assert workflow.success
+         |        |
+         workflow false
+
+  Nextflow stdout:
+
+  ERROR ~ No such file or directory: /workspaces/training/side-quests/nf-test/.nf-test/tests/693ba951a20fec36a5a9292ed1cc8a9f/greetings.csv
+
+   -- Check '/workspaces/training/side-quests/nf-test/.nf-test/tests/693ba951a20fec36a5a9292ed1cc8a9f/meta/nextflow.log' file for details
+  Nextflow stderr:
+
+  Nextflow 24.10.4 is available - Please consider updating your version to it
+
+
+
+
```

Comment:
```suggestion
```
Delete for clarity and to avoid distractions.

Reply:
Sorry, yes, didn't mean to include